### ISSUE TITLE
Add tests to the project

### ISSUE THREAD

**Issue by oneofmanylukes** [andrewyancey/ImageManipulation#9]
Maybe some tests would be a good idea, because then you can see if your changes break anything or have other unexpected effects.

**Comment by Aleksbgbg** [comment]
If the codebase is to be tested, it must first be testable. For the codebase to be testable, we must refactor as per issue #4.

**Comment by oneofmanylukes** [comment]
Sounds as it would be a good idea to refactor then.

**Comment by andrewyancey** [comment]
It’s already started it just needs to be merged into dev

**Comment by andrewyancey** [comment]
Should we add a separate vs project for the test classes?

**Comment by oneofmanylukes** [comment]
Isn't that the way it's usually done? As far as I can tell

**Comment by andrewyancey** [comment]
Yes but I believe it's also possible to do it in the same project. Just curious if you guys wanted the extra overhead of a second project.

Alternatively, I think we could have a testing branch so that only the people who want to run the tests have to pull them.

**Comment by Aleksbgbg** [comment]
Testing branch is a fair point, but it will require more synchronisation.

I don't think the second project adds any significant overhead - in fact, it prevents us from compiling test code into the release build.

**Comment by oneofmanylukes** [comment]
I would also go with a second project for tests

**Comment by andrewyancey** [comment]
Alright let's go with the second project then.
